src/ripple/rpc/handlers/LedgerEntry.cpp

```diff
@@ -621,18 +622,24 @@ doLedgerEntry(RPC::JsonContext& context)
             {
                 uNodeIndex = beast::zero;
                 auto const& oracle = context.params[jss::oracle];
-                auto const documentID =
-                    oracle[jss::oracle_document_id].isConvertibleTo(
-                        Json::ValueType::uintValue)
-                    ? std::make_optional(
-                          oracle[jss::oracle_document_id].asUInt())
-                    : std::nullopt;
+                auto const documentID = [&]() -> std::optional<std::uint32_t> {
+                    auto const& id = oracle[jss::oracle_document_id];
+                    if (id.isConvertibleTo(Json::ValueType::uintValue))
+                        return std::make_optional(id.asUInt());
+                    else if (id.isString())
+                    {
+                        std::uint32_t v;
+                        if (beast::lexicalCastChecked(v, id.asString()))
+                            return std::make_optional(v);
+                    }
+                    return std::nullopt;
+                }();
```

Comment:
This change is to accept IDs passed as strings? Is there a test?

Reply:
Updated ledger entry unit-test.